Dockerfile

```diff
@@ -0,0 +1,40 @@
+# This Dockerfile builds an ruby environment for jekyll that empowers
+# making updates to the accumulo website without requiring the dev
+# to maintain a local ruby development environment.
+
+FROM ruby:2.7.8-slim-bullseye as base
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    git \
+    curl \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /site
+
+
+# Copy over the Gemfiles so that all build dependencies are installed
+# during build vs at runtime.
+
+COPY Gemfile /site/Gemfile
+COPY Gemfile.lock /site/Gemfile.lock
+
+RUN gem update --system && bundle install && gem cleanup
+
+ENV HOST=0.0.0.0
+ENV PORT=4000
+
+EXPOSE $PORT
+
+CMD bundle exec jekyll serve --force-polling -H $HOST -P $PORT
+
+# Create a separate validation container for testing tools that
+# can be used to validate the rendered HTML code.
+
+FROM base
```

Comment:
Should `base` be given a more specific name, in case somebody's Docker environment already has a container with that name, or does Docker already provide enough disambiguation for the name of running containers?

Reply:
base is only relative to this Dockerfile since we've defined what `base` means earlier in the file. 
`FROM ruby:2.7.8-slim-bullseye as base` 

Think of it as a locally scoped variable that will cease to exist once the `docker build` command completes. 
We could make it more explicit like `ruby-base` or `dev-base` but that doesn't provide benefit outside of better naming.